### PR TITLE
feat(core): Multiple token verifiers

### PR DESCRIPTION
**Pull request by javiertoledo**
## Description

This change allows using more than one token verifier for a Booster application. This is useful when your application needs to verify tokens coming from different JWT issuers.

## Changes

Token verifier configuration now accepts an array of `TokenVerifierConfiguration` objects:

```typescript
// From a typical `config/config.ts` file:
Booster.configure('production', (config: BoosterConfig): void => {
  config.appName = 'my-app'
  config.providerPackage = '@boostercloud/framework-provider-aws'
  config.tokenVerifiers = [
    // Accepts tokens signed with a local certificate (For instance, tokens generated for confirmation emails)
    { 
      issuer: 'booster',
      publicKey: fs.readFileSync(path.join(__dirname, '..', '..', 'assets', 'certs', 'public.key'), 'utf8'),
      rolesClaim: 'booster:role',
    },
    // Also accepts tokens signed by Firebase
    {
      jwksUri: 'https://myapp.firebase.com/.well-known/jwks.json',
      issuer: 'https://securetoken.google.com/myapp',
      rolesClaim: 'firebase:groups',
    }
  ]
})
```

## Checks
- [x] Project Builds
- [x] Project passes tests and checks
- [x] Updated documentation accordingly

## Additional information

This PR also deprecates implicit load of default token verifier configuration from environment variables, as it's preferred that the user explicitly sets these options (They can still load the configuration from environment variables)

### CODE REVIEW

**Review by gonzalogarciajaubert — APPROVED** [review]
LGTM

**Review by NickSeagull — APPROVED** [review]

File: docs/chapters/04_features.md [hunk 1]
```diff
@@ -141,13 +141,14 @@ In that way, you can use different authentication providers, like Auth0, Firebas
...
 ### JWT Configuration
 
-In order to use the JWT authorization you will need to set a `tokenVerifier` property which contains the following properties:
+In order to use the JWT authorization you will need to configure at least one token verifier. To do this you can add a `TokenVerifierConfig` entry to the `tokenVerifiers` property which contains the following properties:
 
-- jwksUri: Public uri with the public keys the auth server used to sign in the JWT tokens, commonly known as a key sets.
-- issuer: Identifies the principal that issued the JWT tokens.
-- rolesClaim: Field where provider contains the token. As an example Cognito uses `cognito:groups`.
+- `jwksUri`: URI pointing to a [JWKS](https://datatracker.ietf.org/doc/html/rfc7517#section-4) object containing the public keys provided by the auth provider that signed the JWT tokens.
+- `publicKey`: Alternatively, you can manually provide a public key using this parameter.
```
Comment (alvaroloes) [inline docs/chapters/04_features.md:147]: Thanks for adding this missing piece of doc.

File: packages/framework-core/src/booster-token-verifier.ts [hunk 2]
```diff
@@ -80,3 +78,22 @@ export class BoosterTokenVerifier {
...
+  }
+
+  public async verify(token: string): Promise<UserEnvelope> {
+    const results = await Promise.allSettled(
+      this.tokenVerifierClients.map((tokenVerifierClient) => tokenVerifierClient.verify(token))
+    )
+    const winner = results.find((result) => result.status === 'fulfilled')
+    if (winner) return Promise.resolve((winner as PromiseFulfilledResult<UserEnvelope>).value)
+    return Promise.reject(new Error(results.map((result) => (result as PromiseRejectedResult).reason).join('\n')))
+  }
```
Comment (alvaroloes) [inline packages/framework-core/src/booster-token-verifier.ts:98]: 👏

**Review by alvaroloes — APPROVED** [review]
🚀

**Pull request merged**